gm4_auto_crafting/beet.yaml

```diff
@@ -3,7 +3,9 @@ name: Auto Crafting
 version: 1.2.X
 
 data_pack:
-  load: .
+  load: 
+    - .
+    - data/gm4_custom_crafters/loot_tables/items/custom_crafter.json: ../lib_custom_crafters/data/gm4_custom_crafters/loot_tables/items/custom_crafter.json
```

Comment:
We had a discussion in DMs about this. The `lib_custom_crafter` include happens after the whole module-level build (the module builds, then each library in turn), so the file is not included until after the guidebook processing completes. 

This comes from the guidebook needing to read a loot table (which we're using as an item definition) in order to display it, which when that item comes from another module needs to be added to the load. We'd have to do the same if say orbis wanted to display a heart canister. I think it makes more sense in the non-library case to indicate a "dependancy" from another pack in this way

Reply:
I agree it feels wrong, but Special said it was the best way to do it. The plugin includes it as `gm4_custom_crafters-3.1/loot_tables/...` and I think the guidebook script currently runs before the cc plugin is run

Reply:
I see. We had a discussion during the beet 2 phase whether to include the libraries before or after the module, I guess here is where it causes problems.

Is there a possibility in running the guidebook stuff later, after all the libraries have been added and other pipeline stuff has run? Other modules may want to generate loot tables or recipes like this in the pipeline (in a custom plugin after `gm4.plugins.extend.module`). So I think moving the guidebook processing order back a bit is good either way.

Reply:
I am now realizing that a loot table used only as an item definition (for guidebook display or custom crafter recipe generation) probably shouldn't be included in the output build. It could cause strangeness where the same file is in multiple packs being affected by load order :/
Maybe theres another way we can mount the loot table thats better than this

Reply:
> I am now realizing that a loot table used only as an item definition (for guidebook display or custom crafter recipe generation) probably shouldn't be included in the output build. It could cause strangeness where the same file is in multiple packs being affected by load order :/
That's what I said :/
>  I don't think it makes sense to make the (normal) datapack developer to remember to mount a loot table to display an item.

Reply:
The original intention with `extend.module` was that it would be the last thing pipeline step to run before libraries. It handles all plugins that are shared by all modules and finished with mecha (which nothing should run after). A module looking to generate assets should happen before `extend.module`

Libraries being included last was similarly to ensure that the module's plugins didn't unintentionally process library files, such as find-replace operations, and could in the future be somewhat cached to improve build time.

Reply:
I don't think this really makes sense, nor solves the current problem. Doing guidebook at the very end would mean that its operating on post-mecha files, which would require a rewrite of the skin-average-color-reading functionality, and in the future, make reading CMD values a bit obtuse. This would also mount the library files under their versioned namespace, which we don't want to reference in the page definition (`{"insert":"gm4_custom_crafters-3.1:item/custom_crafter"}` is bad). It also wouldn't solve wanting to display or use an item from another module in a recipe.

(Also can we move to discord lol. )